tests/mock_crust/tunnel.rs

```diff
@@ -271,6 +272,52 @@ fn tunnel_node_blocked() {
     assert!(tunnel_node_index != unwrap!(locate_tunnel_node(&nodes, PeerId(2), PeerId(3))));
 }
 
+
+#[test]
+fn tunnel_node_dropped() {
+    let min_section_size = 5;
+    let network = Network::new(min_section_size, None);
+    network.block_connection(Endpoint(2), Endpoint(3));
+    network.block_connection(Endpoint(3), Endpoint(2));
+    let mut nodes = create_connected_nodes(&network, min_section_size);
+    let _ = poll_all(&mut nodes, &mut []);
+    verify_invariant_for_all_nodes(&nodes);
+
+    let tunnel_node_index = unwrap!(locate_tunnel_node(&nodes, PeerId(2), PeerId(3)));
+    assert_eq!(1, tunnel_node_index);
+    let _ = nodes.remove(tunnel_node_index);
+
+    poll_and_resend(&mut nodes, &mut []);
+    expect_any_event!(nodes[1], Event::NodeAdded(..) if true);
+    expect_any_event!(nodes[2], Event::NodeAdded(..) if true);
```

Comment:
Those `if true` look unnecessary - if the macro needs them for some reason, a comment would be nice :)